cunumeric/utils.py

```diff
@@ -55,54 +90,33 @@ def find_last_user_frames(top_only=True):
     return "|".join(get_line_number_from_frame(f) for f in frames)
 
 
-# These are the dtypes that we currently support for cuNumeric
-def is_supported_dtype(dtype):
+def is_supported_dtype(dtype: Any) -> bool:
     if not isinstance(dtype, np.dtype):
         raise TypeError("expected a NumPy dtype")
-    base_type = dtype.type
-    if (
-        base_type == np.float16
-        or base_type == np.float32
-        or base_type == np.float64
-        or base_type == float
-    ):
-        return True
-    if (
-        base_type == np.int16
-        or base_type == np.int32
-        or base_type == np.int64
-        or base_type == int
-    ):
-        return True
-    if (
-        base_type == np.uint16
-        or base_type == np.uint32
-        or base_type == np.uint64
-    ):
-        return True
-    if base_type == np.bool_ or base_type == bool:
-        return True
-    return False
-
-
-def calculate_volume(shape):
+    return dtype.type in _SUPPORTED_DTYPES
+
+
+def calculate_volume(shape: tuple[int, ...]) -> int:
     if shape == ():
         return 0
     return reduce(lambda x, y: x * y, shape)
 
 
-def get_arg_dtype(dtype):
+def get_arg_dtype(dtype: np.dtype[Any]) -> np.dtype[Any]:
     return np.dtype(
         [("arg", np.int64), ("arg_value", dtype)],
         align=True,
     )
 
 
-def get_arg_value_dtype(dtype):
-    return dtype.fields["arg_value"][0].type
+def get_arg_value_dtype(dtype: np.dtype[Any]) -> Any:
```

Comment:
Feels like the return type should be also `np.dtype[Any]`, but I suspect you already tried it. Is there a reason that that didn't work?

Reply:
`"arg_value"` is metadata set up by us unless I am mistaken, I'm not sure of a good way to link that up statically. We could certainly add a cast here on the return if you want to *assert* that that will be the return.

Reply:
I went ahead and added the cast and tightened up the return value in https://github.com/nv-legate/cunumeric/pull/268/commits/bb62e6c25097a82a3430957e488833e21962f430

Reply:
Ah, no, this won't work:
```
  File "/home/bryan/work/cunumeric/cunumeric/utils.py", line 114, in get_arg_value_dtype
    return cast(np.dtype[Any], dt)
TypeError: 'numpy._DTypeMeta' object is not subscriptable
```

It's possible a newer numpy version will work I don't know offhand though

Reply:
WAR with cast to `Any` internally instead, for now https://github.com/nv-legate/cunumeric/pull/268/commits/2edb9c82c21bfac9900950fb36cde732fa826f6b